### ISSUE TITLE
Feature: Validate Settings Input before training start

### ISSUE THREAD

**Issue by JensBlack** [sgoldenlab/simba#21]
**Is your feature request related to a problem? Please describe.**
The Settings for training a machine model in SimBa have multiple options that present a predefined Value (e.g. NaN) when they are toggled. If you accept invalid settings (such as NaN) there is no feedback for the user but training the model will result in an error (often at the end of training). 
See below for an example:

```
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 12 concurrent workers.
[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    4.9s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   26.0s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:  1.0min
[Parallel(n_jobs=-1)]: Done 776 tasks      | elapsed:  1.9min
[Parallel(n_jobs=-1)]: Done 1226 tasks      | elapsed:  3.0min
[Parallel(n_jobs=-1)]: Done 1776 tasks      | elapsed:  4.4min
[Parallel(n_jobs=-1)]: Done 2000 out of 2000 | elapsed:  5.0min finished
Exception in thread Thread-1:
Traceback (most recent call last):
  File "D:\Anaconda\envs\simba-gpu\lib\threading.py", line 926, in _bootstrap_inner
    self.run()
  File "D:\Anaconda\envs\simba-gpu\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "simba.py", line 3737, in trainsinglemodel
    trainmodel2(self.projectconfigini)
  File "C:\Users\schwa\PycharmProjects\simba\simba\train_model_2.py", line 259, in trainmodel2
    shuffle_splits = config.getint('create ensemble settings', 'LearningCurve_shuffle_k_splits')
  File "D:\Anaconda\envs\simba-gpu\lib\configparser.py", line 818, in getint
    fallback=fallback, **kwargs)
  File "D:\Anaconda\envs\simba-gpu\lib\configparser.py", line 808, in _get_conv
    **kwargs)
  File "D:\Anaconda\envs\simba-gpu\lib\configparser.py", line 802, in _get
    return conv(self.get(section, option, **kwargs))
ValueError: invalid literal for int() with base 10: 'NaN'
```

I assumed that there was a default value that would be taken if i leave the textbox 'NaN'. 

**Describe the solution you'd like**
This is a user created  issue (and i won't do it again), but it might be worthwhile to either validate the input when saving the settings (e.g. comparing if it is 'int', 'float', etc.). Then if it is invalid, setting the value to default (while telling the user) or making the user choose again. This would save some users (like me) from their own ignorance and others from wasting time training a machine model that will ultimately not be saved, because the validation process causes an error and everything is gone.

**Describe alternatives you've considered**
An alternative would be to validate the settings before starting the training. Simply throwing out an error (e.g. "ValueError: Some of your parameters are invalid. Consider reading the readme files again.")

Another alternative would be to check user input when pressing the save button. Similiar to a contact formular on a website. I guess this would take much more time (as it's related to gui programming) though.

**Additional context**
I would not document this kind of request if your overall goal wasn't to create a tool that can be used without any (or without a lot of) programming experience. SimBa is already simplifying a complicated process so much and it is doing great work!
